library/Faktory/Connection.hs

```diff
@@ -82,6 +83,44 @@ connect ConnectionInfo {..} = bracketOnError open connectionClose pure
       , connectionUseSocks = Nothing
       }
 
+  connectTo' :: ConnectionContext -> ConnectionParams -> IO Connection
+  connectTo' cg cParams =
+    bracketOnError
+      (resolve' (connectionHostname cParams) (connectionPort cParams))
+      (close . fst)
+      ( \(h, _) ->
+        connectFromSocket cg h cParams
+      )
+
+  -- This is copy and pasted from the connectTo implement. The only change is
+  -- adding a keep alive socket option.
+  -- see: https://hackage.haskell.org/package/connection-0.3.1/docs/src/Network.Connection.html#connectTo
+  resolve' :: String -> PortNumber -> IO (Socket, SockAddr)
+  resolve' host port = do
+      let hints = defaultHints { addrFlags = [AI_ADDRCONFIG], addrSocketType = Stream }
+      addrs <- getAddrInfo (Just hints) (Just host) (Just $ show port)
+      firstSuccessful $ map tryToConnect addrs
+    where
+      tryToConnect addr =
+          bracketOnError
+            (socket (addrFamily addr) (addrSocketType addr) (addrProtocol addr))
+            close
+            (\sock -> do
+              setSocketOption sock KeepAlive 1
```

Comment:
It seems like we could try to upstream some kind of `connectToWith` that takes a function to call at this point, then redefine `connectTo` as,

```hs
connectTo = connectToWith $ const pure ()
```

And we could use `connectToWith $ \sock -> setSocketOption sock KeepAlive 1`.

Reply:
On second-thought, extending `ConnectionParams` to have some kind of "setup socket" field would probably be cleaner. Though since that type is exported by the module that change would be a major version bump (whereas what I described above would not).

Reply:
There is an old issue open speaking about this

https://github.com/vincenthz/hs-connection/issues/48

@vincenthz has been inactive in that repo since 2019, so it is possible `connection` is unmaintained.

Reply:
Just to clarify, I was suggestion we still merge this work, but seek to remove it later by pushing upstream.

Given the inactivity though, maybe we just accept it has to live here?